test/Microsoft.AspNetCore.Mvc.IntegrationTests/ComplexTypeModelBinderIntegrationTest.cs

```diff
@@ -2701,7 +2701,7 @@ private class LocationInfo
 
             entry = Assert.Single(modelState, e => e.Key == "Info.Value.GpsCoordinates").Value;
             Assert.Equal("10,20", entry.AttemptedValue);
-            Assert.Equal(new[] { "10", "20" }, entry.RawValue);
+            Assert.Equal("10,20", entry.RawValue);
```

Comment:
Wait. This implies something is changing that shouldn't be. I'm not that worried about `RawValue` per se. But, the "are we assigning to a non-collection" checks were intended to keep the behaviour as close as possible to the old ways. Why is this acting differently?

Reply:
This is because HMB used to do this earlier https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/HeaderModelBinder.cs#L97